tests/when_working_with_raw_stream_metadata.php

```diff
@@ -13,232 +13,203 @@
 
 namespace ProophTest\EventStoreClient;
 
-use function Amp\call;
-use Amp\PHPUnit\AsyncTestCase;
-use Amp\Promise;
-use Closure;
 use Generator;
-use Prooph\EventStore\Async\EventStoreConnection;
 use Prooph\EventStore\Exception\StreamDeleted;
 use Prooph\EventStore\Exception\WrongExpectedVersion;
 use Prooph\EventStore\ExpectedVersion;
 use Prooph\EventStore\RawStreamMetadataResult;
-use ProophTest\EventStoreClient\Helper\TestConnection;
 use ProophTest\EventStoreClient\Helper\TestEvent;
 
-class when_working_with_raw_stream_metadata extends AsyncTestCase
+class when_working_with_raw_stream_metadata extends EventStoreConnectionTestCase
 {
     private string $stream;
-    private EventStoreConnection $conn;
 
-    private function execute(Closure $function): Promise
+    protected function setUp(): void
     {
-        return call(function () use ($function): Generator {
-            $this->stream = self::class . '\\' . $this->getName();
-            $this->conn = TestConnection::create();
-            yield $this->conn->connectAsync();
+        parent::setUp();
 
-            yield from $function();
-        });
+        $this->stream = self::class . '\\' . $this->getName();
     }
 
     /**
      * @test
      */
     public function setting_empty_metadata_works(): Generator
     {
-        yield $this->execute(function (): Generator {
-            yield $this->conn->setStreamMetadataAsync(
-                $this->stream,
-                ExpectedVersion::NO_STREAM
-            );
-
-            $meta = yield $this->conn->getRawStreamMetadataAsync($this->stream);
-            \assert($meta instanceof RawStreamMetadataResult);
-            $this->assertEquals($this->stream, $meta->stream());
-            $this->assertFalse($meta->isStreamDeleted());
-            $this->assertEquals(0, $meta->metastreamVersion());
-            $this->assertEquals('', $meta->streamMetadata());
-        });
+        yield $this->connection->setStreamMetadataAsync(
+            $this->stream,
+            ExpectedVersion::NO_STREAM
+        );
+
+        $meta = yield $this->connection->getRawStreamMetadataAsync($this->stream);
+        \assert($meta instanceof RawStreamMetadataResult);
+        $this->assertEquals($this->stream, $meta->stream());
+        $this->assertFalse($meta->isStreamDeleted());
+        $this->assertEquals(0, $meta->metastreamVersion());
+        $this->assertEquals('', $meta->streamMetadata());
```

Comment:
assertSame?

Btw. I tend to use `self::assert*` instead since these methods are actually static.

Reply:
see https://phpunit.readthedocs.io/en/9.0/assertions.html

Quote: `The short answer is: there is no right way. And there is no wrong way, either. It is a matter of personal preference.`

Reply:
I see. Okay, it's fine if you prefer that.

But still, why are you using assertEquals instead of assertSame?